Rubberduck.Core/UI/Converters/DeclarationToMemberSignatureConverter.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Windows;
+using System.Windows.Data;
+
+namespace Rubberduck.UI.Converters
+{
+    public class DeclarationToMemberSignatureConverter : IValueConverter
+    {
+        public class Parameter
+        {
+            public string ParamAccessibility { get; set; }
+            public string ParamName { get; set; }
+            public string ParamType { get; set; }
+
+            public override string ToString()
+            {
+                return $"{ParamAccessibility} {ParamName} As {ParamType}";
+            }
+        }
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var typedValue = (Declaration)value;
+            return FullMemberSignature(typedValue);
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+
+        private string FullMemberSignature(Declaration member)
+        {
+            var signature = $"{GetMethodType(member)} {member.IdentifierName}({string.Join(", ", GetMemberParameters(member))})";
+
+            return member.AsTypeName == null ? signature : $"{signature} As {member.AsTypeName}";
+        }
+
+        private List<Parameter> GetMemberParameters(Declaration member)
+        {
+            var parameters = new List<Parameter>();
+
+            if (member is IParameterizedDeclaration memberWithParams)
+            {
+                parameters = memberWithParams.Parameters
+                    .OrderBy(o => o.Selection.StartLine)
+                    .ThenBy(t => t.Selection.StartColumn)
+                    .Select(p => new Parameter
+                    {
+                        ParamAccessibility =
+                            ((VBAParser.ArgContext)p.Context).BYVAL() != null ? Tokens.ByVal : Tokens.ByRef,
+                        ParamName = p.IdentifierName,
+                        ParamType = p.AsTypeName
+                    })
+                    .ToList();
+            }
+            else
+            {
+                return new List<Parameter>();
+            }
+
+            if (GetMethodType(member) == "Property Get")
```

Comment:
Feels wrong when the method returns `$"{Tokens.Property} {Tokens.Get}"`, would prefer to have string interpolation here too... but why can we assume `parameters.Count` is >= 1 here? Most `Property Get` members are parameterless... :confused: